tests/zkevm/test_worst_compute.py

```diff
@@ -952,6 +952,49 @@ def test_worst_mod(
     )
 
 
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("opcode", [Op.MLOAD, Op.MSTORE, Op.MSTORE8])
+@pytest.mark.parametrize("offset", [0, 1, 31])
+@pytest.mark.parametrize("offset_initialized", [True, False])
+@pytest.mark.parametrize("big_memory_expansion", [True, False])
+def test_worst_memory_access(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    opcode: Op,
+    offset: int,
+    offset_initialized: bool,
+    big_memory_expansion: bool,
+):
+    """Test running a block with as many memory access instructions as possible."""
+    env = Environment()
+
+    mem_exp_code = Op.MSTORE8(10 * 1024, 1) if big_memory_expansion else Bytecode()
+    offset_set_code = Op.MSTORE(offset, 43) if offset_initialized else Bytecode()
+    code_prefix = mem_exp_code + offset_set_code + Op.PUSH1(42) + Op.PUSH1(offset) + Op.JUMPDEST
```

Comment:
The `Op.PUSH1(42)` is not necessary for the `MLOAD` case, but since it's `code_prefix` shouldn't be a big deal.